llvm/lib/Transforms/Utils/CodeExtractor.cpp

```diff
@@ -1579,6 +1579,11 @@ static void fixupDebugInfoPostExtraction(Function &OldFunc, Function &NewFunc,
       DebugIntrinsicsToDelete.push_back(DVI);
       continue;
     }
+    if (auto *DAI = dyn_cast<DbgAssignIntrinsic>(DVI);
+        DAI && IsInvalidLocation(DAI->getAddress())) {
+      DebugIntrinsicsToDelete.push_back(DVI);
```

Comment:
Can you add a comment explaining why?

Reply:
There is a comment a few lines above, but sadly github doesn't show it. Does this address the concern?

Reply:
Added a comment focusing on why dbg assign is different from the other intrinsics.